packages/service-provider-server/src/compass/compass-service-provider.ts

```diff
@@ -24,6 +26,7 @@ class CompassServiceProvider extends CliServiceProvider {
   constructor(mongoClient: MongoClient, uri?: string) {
     super(mongoClient, uri);
     this.platform = ReplPlatform.Compass;
+    modifyBson(mongodb, this.platform);
```

Comment:
This changes the platform, since if we didn't override it it would have `ReplPlatform.CLI` instead of `ReplPlatform.Compass` as the argument to modifyBson. However that's assuming that the method actually needs to do different things depending on the platform - if we can find a solution that will work with all 3 platforms then we don't need the platform argument at all (and will not have the circular dep). I think if I understand your earlier comment, we don't actually need the platform in `modifyBson` right?

Reply:
Yep browserify should do that for us